pkg/operation/botanist/component/extensions/dnsrecord/dnsrecord_test.go

```diff
@@ -219,105 +223,459 @@ var _ = Describe("DNSRecord", func() {
+					ObjectMeta: metav1.ObjectMeta{
+						Name:            name,
+						Namespace:       namespace,
+						ResourceVersion: "2",
+						Annotations: map[string]string{
+							v1beta1constants.GardenerTimestamp: now.UTC().String(),
+						},
+					},
+					Spec: dns.Spec,
+				}))
+			})
 
-		It("should update the timestamp annotation if ReconcileOnce is true and the DNSRecord is found", func() {
-			values.ReconcileOnce = true
-			dnsRecord = dnsrecord.New(log, c, values, dnsrecord.DefaultInterval, dnsrecord.DefaultSevereThreshold, dnsrecord.DefaultTimeout)
-			delete(dns.Annotations, v1beta1constants.GardenerOperation)
-			// set old timestamp (e.g. added on creation / earlier Deploy call)
-			metav1.SetMetaDataAnnotation(&dns.ObjectMeta, v1beta1constants.GardenerTimestamp, now.UTC().Add(-time.Second).String())
-			Expect(c.Create(ctx, dns)).To(Succeed())
+			Context("when the referenced secret name changes", func() {
+				BeforeEach(func() {
+					secretName = "testsecret"
+				})
 
-			Expect(dnsRecord.Deploy(ctx)).To(Succeed())
+				It("should prefix the referenced secret of existing DNSRecords with 'dnsrecord-' when ReconcileOnValueChangeOnly is true", func() {
+					delete(dns.Annotations, v1beta1constants.GardenerOperation)
+					// set old timestamp (e.g. added on creation / earlier Deploy call)
+					metav1.SetMetaDataAnnotation(&dns.ObjectMeta, v1beta1constants.GardenerTimestamp, now.UTC().Add(-time.Second).String())
+					Expect(c.Create(ctx, dns)).To(Succeed())
+
+					values.SecretName = "dnsrecord-testsecret"
+					Expect(dnsRecord.Deploy(ctx)).To(Succeed())
+
+					deployedDNS := &extensionsv1alpha1.DNSRecord{}
+					err := c.Get(ctx, client.ObjectKey{Name: name, Namespace: namespace}, deployedDNS)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(deployedDNS).To(DeepEqual(&extensionsv1alpha1.DNSRecord{
+						TypeMeta: metav1.TypeMeta{
+							APIVersion: extensionsv1alpha1.SchemeGroupVersion.String(),
+							Kind:       extensionsv1alpha1.DNSRecordResource,
+						},
+						ObjectMeta: metav1.ObjectMeta{
+							Name:            name,
+							Namespace:       namespace,
+							ResourceVersion: "2",
+							Annotations: map[string]string{
+								v1beta1constants.GardenerOperation: v1beta1constants.GardenerOperationReconcile,
+								v1beta1constants.GardenerTimestamp: now.UTC().String(),
+							},
+						},
+						Spec: extensionsv1alpha1.DNSRecordSpec{
+							DefaultSpec: extensionsv1alpha1.DefaultSpec{
+								Type: extensionType,
+							},
+							SecretRef: corev1.SecretReference{
+								Name:      "dnsrecord-testsecret",
+								Namespace: namespace,
+							},
+							Zone:       pointer.String(zone),
+							Name:       dnsName,
+							RecordType: extensionsv1alpha1.DNSRecordTypeA,
+							Values:     []string{address},
+							TTL:        pointer.Int64(ttl),
+						},
+					}))
+				})
+			})
+			Context("when dnsrecord values change", func() {
+				It("should reconcile when DNSName changes", func() {
```

Comment:
This context contains a lot of different test cases for the different `Values` fields that may have changed. While it's extensive, most of the code is heavily duplicated. Could we perhaps eliminate this duplication by either capturing some of it in variables or functions, using `DescribeTable`, or both?

Reply:
done.